Modules/cjkcodecs/multibytecodec.c

```diff
@@ -825,8 +826,13 @@ encoder_encode_stateful(MultibyteStatefulEncoderContext *ctx,
     if (inpos < datalen) {
         if (datalen - inpos > MAXENCPENDING) {
             /* normal codecs can't reach here */
-            PyErr_SetString(PyExc_UnicodeError,
-                            "pending buffer overflow");
+            excobj = PyObject_CallFunction(PyExc_UnicodeEncodeError,
+                                           "ssnns",
+                                           ctx->codec->encoding,
+                                           PyUnicode_AsUTF8(inbuf),
```

Comment:
Do we need copy `inbuf`? If no, use just `inbuf` with `O` format.

Reply:
Done 👍 I think I tried this originally, but I must have been doing something different because it seems to work fine now.